bundles/org.openhab.binding.ferroamp/src/main/java/org/openhab/binding/ferroamp/internal/FerroampMqttCommunication.java

```diff
@@ -0,0 +1,1028 @@
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonObject;
+
+/**
+ * The {@link FerroampMqttCommunication} is responsible for communication with the Ferroamp-systems Mqtt-broker.
+ *
+ * @author Örjan Backsell - Initial contribution
+ *
+ */
+
+@NonNullByDefault
+public class FerroampMqttCommunication implements MqttMessageSubscriber {
+    static String[] ehubChannelsUpdateValues = new String[0];
+    static String[] ssoS1ChannelsUpdateValues = new String[0];
+    static String[] ssoS2ChannelsUpdateValues = new String[0];
+    static String[] ssoS3ChannelsUpdateValues = new String[0];
+    static String[] ssoS4ChannelsUpdateValues = new String[0];
+    static String[] esoChannelsUpdateValues = new String[0];
+    static String[] esmChannelsUpdateValues = new String[0];
+
+    static boolean isSsoChecked = false;
+    static String ssoS1IdCheck = "";
+    static String ssoS2IdCheck = "";
+    static String ssoS3IdCheck = "";
+    static String ssoS4IdCheck = "";
+
+    private final static Logger logger = LoggerFactory.getLogger(FerroampMqttCommunication.class);
+
+    public FerroampMqttCommunication(Thing thing) {
+        super();
+    }
+
+    // Handles request topic
+    static void sendPublishedTopic(String payload, FerroampConfiguration ferroampConfig) {
+        MqttBrokerConnection localConfigurationConnection = FerroampHandler.getFerroampConnection();
+        Objects.requireNonNull(localConfigurationConnection,
+                "MqttBrokerConnection localConfigurationConnection cannot be null");
+        localConfigurationConnection.start();
+        localConfigurationConnection.setCredentials(ferroampConfig.userName, ferroampConfig.password);
+        localConfigurationConnection.publish(FerroampBindingConstants.REQUEST_TOPIC, payload.getBytes(), 1, false);
+    }
+
+    // Handles respective topic type
+    void getSubscribedTopic(String topic, FerroampConfiguration ferroampConfig) {
+        MqttBrokerConnection localSubscribeConnection = FerroampHandler.getFerroampConnection();
+        Objects.requireNonNull(localSubscribeConnection,
+                "MqttBrokerConnection localSubscribeConnection cannot be null");
+        if ("ehubTopic".equals(topic)) {
+            localSubscribeConnection.subscribe(FerroampBindingConstants.EHUB_TOPIC, this);
+        }
+        if ("ssoTopic".equals(topic)) {
+            localSubscribeConnection.subscribe(FerroampBindingConstants.SSO_TOPIC, this);
+        }
+        if ("esoTopic".equals(topic)) {
+            localSubscribeConnection.subscribe(FerroampBindingConstants.ESO_TOPIC, this);
+        }
+        if ("esmTopic".equals(topic)) {
+            localSubscribeConnection.subscribe(FerroampBindingConstants.ESM_TOPIC, this);
+        }
+    }
+
+    // Capture actual Json-topic message
+    @Override
+    public void processMessage(String topic, byte[] payload) {
+        if ("extapi/data/ehub".equals(topic)) {
```

Comment:
You should use the bindingconstants for these strings as they are already available.

Reply:
20250319
Changed to bindingconstants, Please see line 71 - 103. I have made the change for each topic